device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -117,26 +117,17 @@ public Object getReceiveThreadLock()
 
     public boolean hasMessagesToSend()
     {
-        synchronized (sendThreadLock)
```

Comment:
This method is only called by the send thread when it has grabbed this lock. No need to try to grab it again here

Reply:
Same here and below